managed/services/backup/backup_service.go

```diff
@@ -84,94 +86,115 @@ func (s *Service) PerformBackup(ctx context.Context, params PerformBackupParams)
-				return err
+				if err = services.CheckMongoDBBackupPreconditions(tx.Querier, params.Mode, svc.Cluster, svc.ServiceID, params.ScheduleID); err != nil {
+					return err
+				}
+
+				// For PITR backups reuse existing artifact if it's present.
+				if params.Mode == models.PITR {
+					artifact, err = models.FindArtifactByName(tx.Querier, name)
+					if err != nil && !errors.Is(err, models.ErrNotFound) {
+						return err
+					}
+				}
+
+			case models.PostgreSQLServiceType,
+				models.ProxySQLServiceType,
+				models.HAProxyServiceType,
+				models.ExternalServiceType:
+				return status.Errorf(codes.Unimplemented, "Unimplemented service: %s", svc.ServiceType)
+			default:
+				return status.Errorf(codes.Unknown, "Unknown service: %s", svc.ServiceType)
 			}
 
-			// For PITR backups reuse existing artifact if it's present.
-			if params.Mode == models.PITR {
-				artifact, err = models.FindArtifactByName(tx.Querier, name)
-				if err != nil && !errors.Is(err, models.ErrNotFound) {
+			if artifact == nil {
+				if artifact, err = models.CreateArtifact(tx.Querier, models.CreateArtifactParams{
+					Name:       name,
+					Vendor:     string(svc.ServiceType),
+					DBVersion:  dbVersion,
+					LocationID: locationModel.ID,
+					ServiceID:  svc.ServiceID,
+					DataModel:  params.DataModel,
+					Mode:       params.Mode,
+					Status:     models.PendingBackupStatus,
+					ScheduleID: params.ScheduleID,
+				}); err != nil {
+					return err
+				}
+			} else {
+				if artifact, err = models.UpdateArtifact(tx.Querier, artifact.ID, models.UpdateArtifactParams{
+					Status: models.BackupStatusPointer(models.PendingBackupStatus),
+				}); err != nil {
 					return err
 				}
 			}
 
-		case models.PostgreSQLServiceType,
-			models.ProxySQLServiceType,
-			models.HAProxyServiceType,
-			models.ExternalServiceType:
-			return status.Errorf(codes.Unimplemented, "Unimplemented service: %s", svc.ServiceType)
-		default:
-			return status.Errorf(codes.Unknown, "Unknown service: %s", svc.ServiceType)
-		}
-
-		if artifact == nil {
-			if artifact, err = models.CreateArtifact(tx.Querier, models.CreateArtifactParams{
-				Name:       name,
-				Vendor:     string(svc.ServiceType),
-				DBVersion:  dbVersion,
-				LocationID: locationModel.ID,
-				ServiceID:  svc.ServiceID,
-				DataModel:  params.DataModel,
-				Mode:       params.Mode,
-				Status:     models.PendingBackupStatus,
-				ScheduleID: params.ScheduleID,
-			}); err != nil {
-				return err
-			}
-		} else {
-			if artifact, err = models.UpdateArtifact(tx.Querier, artifact.ID, models.UpdateArtifactParams{
-				Status: models.BackupStatusPointer(models.PendingBackupStatus),
-			}); err != nil {
+			if job, dbConfig, err = s.prepareBackupJob(tx.Querier, svc, artifact.ID, jobType, params.Mode, params.DataModel, params.Retries, params.RetryInterval); err != nil { //nolint:lll
 				return err
 			}
+			return nil
+		})
+
+		if errTX == nil || i >= 30 {
```

Comment:
Can we document this magic number?

Reply:
In general it will be refactored soon. For now just functioning fix.

p.s. commited your suggestion